src/server/scripts/Kalimdor/zone_silithus.cpp

```diff
@@ -1214,9 +1217,29 @@ class go_wind_stone : public GameObjectScript
 
                 void SummonNPC(GameObject* go, Player* player, uint32 npc, uint32 spell)
                 {
-                    player->CastSpell(player, spell);
+                    isSummoning = true;
+                    player->CastSpell(player, spell, true);
                     TempSummon* summons = go->SummonCreature(npc, go->GetPositionX(), go->GetPositionY(), go->GetPositionZ(), player->GetOrientation() - float(M_PI), TEMPSUMMON_TIMED_OR_DEAD_DESPAWN, 10min);
                     summons->CastSpell(summons, SPELL_SPAWN_IN, false);
+
+                    // Add player to SmartAI stored target list 1
+                    ObjectVector primaryTarget;
+                    primaryTarget.push_back(player);
+                    ENSURE_AI(SmartAI, summons->AI())->GetScript()->AddToStoredTargetList(primaryTarget, 1);
+
+                    // Add player's group (if applicable) within 50yds to SmartAI stored target list 2
+                    if (Group* group = player->GetGroup())
+                    {
+                        ObjectVector secondaryTargets;
+                        for (GroupReference* groupRef = group->GetFirstMember(); groupRef != nullptr; groupRef = groupRef->next())
+                            if (Player* member = groupRef->GetSource())
+                                if (member != player && member->IsInMap(summons) and member->GetDistance(summons) < 50.0f)
+                                    secondaryTargets.push_back(member);
+
+                        if (secondaryTargets.size() > 0)
+                            ENSURE_AI(SmartAI, summons->AI())->GetScript()->AddToStoredTargetList(secondaryTargets, 2);
```

Comment:
`ENSURE_AI` will cause a crash if for some reason summon AI was not SAI, to be safe you need to check it
```
if (SmartAI* sai = CAST_AI(SmartAI, summons->AI()))
    sai->GetScript()->AddToStoredTargetList(secondaryTargets, 2);
```

Reply:
Oh, I thought that ENSURE_AI was meant to do exactly that, "Ensure AI" :D . I guess it does not. Will fix, thanks.

Reply:
Yeah not having it as SmartAI triggers an assertion and "crashes" the server. Shouldn't that happen though? Isn't it what asserts are for, to stop execution is something is very wrong, like it is if the unit does not have SmartAI.

Reply:
No, scripts are never supposed to crash the server in case of bad data in database (such as missing scriptnames/ainame assignments)

Reply:
`ENSURE_AI` is supposed to be used only when 100% sure it will never ever not be of the requested type. This is useful to mark cases as "it will never happen" for static analysis reasons and to get an assert at the correct point instead of a null dereference check sometime further down the code.

As a rule of thumb, it's good to check the implementation of methods when using them. See https://github.com/TrinityCore/TrinityCore/blob/a8475e5d70c4a49b3cd9865f5059ac12a333a98c/src/server/game/AI/CoreAI/UnitAI.h#L29-L37 on how `ENSURE_AI` is implemented

Reply:
Yeah triggering an assertion is indeed a nuclear option, I understand now that ENSURE_AI means "absolutely ensure in matters of life and death" and not in db or script errors or such. I added the proper checks and logged errors instead of using assertion.